src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PrisonerIndexService.kt

```diff
@@ -110,10 +113,11 @@ class PrisonerIndexService(
     return indexStatusService.getCurrentIndex()
   }
 
-  private fun checkExistsAndReset(prisonerIndex: SyncIndex) {
+  internal fun checkExistsAndReset(prisonerIndex: SyncIndex) {
     if (checkIfIndexExists(prisonerIndex.indexName)) {
       searchClient.lowLevelClient().performRequest(Request("DELETE", "/${prisonerIndex.indexName}"))
     }
+    await untilCallTo { checkIfIndexExists(prisonerIndex.indexName) } matches { it == false }
```

Comment:
Only used this in test code before - can't think of a reason not to use it in production code?

Reply:
I assume there is a default before it "times out" 10 seconds? Is the default enough???

Reply:
Yes it is 10 seconds.  I think this should be enough - currently it works 99% of the time and the existing "time out" (i.e. the time it makes to request the delete) is a few ms.